extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/InfinispanClientProcessor.java

```diff
@@ -206,6 +207,22 @@ BeanContainerListenerBuildItem build(InfinispanTemplate template, PropertiesBuil
         return new BeanContainerListenerBuildItem(template.configureInfinispan(properties));
     }
 
+    @Record(ExecutionTime.RUNTIME_INIT)
+    @BuildStep
+    void configureRuntimeProperties(InfinispanTemplate template,
+            InfinispanClientConfigRuntime infinispanClientConfigRuntime,
+            PropertiesBuildItem propertiesBuildItem) {
+        Optional<String> serverList = infinispanClientConfigRuntime.serverList;
+        if (!serverList.isPresent()) {
+            // No server list configured
+            return;
+        }
+
+        template.configureRuntimeProperties(infinispanClientConfigRuntime);
+
+        propertiesBuildItem.getProperties().put(ConfigurationProperties.SERVER_LIST, serverList.get());
```

Comment:
I don't think I would read and set the properties at build time, because that's what this line does.

Reply:
this is why it was working for me, replacement of the properties. I mixed things up between Processor and Producer files. Pushing a change